packages/strapi-connector-mongoose/lib/buildQuery.js

```diff
@@ -351,65 +353,87 @@ const buildLookup = ({ model, key, paths }) => {
 /**
  * Build a lookup match expression (equivalent to a SQL join condition)
  * @param {Object} options - Options
- * @param {Object} options.assoc - The association on which is based the ematching xpression
+ * @param {Object} options.assoc - The association on which is based the matching expression
  */
-const buildLookupMatch = ({ assoc }) => {
+const buildLookupMatch = ({ assoc, assocModel, filters = {} }) => {
+  const defaultMatches = [];
+
+  if (hasDraftAndPublish(assocModel) && DP_PUB_STATES.includes(filters.publicationState)) {
+    const dpQuery = populateQueries.publicationState[filters.publicationState];
+
+    if (_.isObject(dpQuery)) {
+      defaultMatches.push(dpQuery);
+    }
+  }
+
   switch (assoc.nature) {
     case 'oneToOne': {
       return [
         {
           $match: {
-            $expr: {
-              $eq: [`$${assoc.via}`, '$$localId'],
-            },
+            $and: defaultMatches.concat({
+              $expr: {
+                $eq: [`$${assoc.via}`, '$$localId'],
+              },
+            }),
           },
         },
       ];
     }
     case 'oneToMany': {
       return {
         $match: {
-          $expr: {
-            $eq: [`$${assoc.via}`, '$$localId'],
-          },
+          $and: defaultMatches.concat({
+            $expr: {
+              $eq: [`$${assoc.via}`, '$$localId'],
+            },
+          }),
         },
       };
     }
     case 'oneWay':
     case 'manyToOne': {
       return {
         $match: {
-          $expr: {
-            $eq: ['$$localAlias', '$_id'],
-          },
+          $and: defaultMatches.concat({
+            $expr: {
+              $eq: ['$$localAlias', '$_id'],
+            },
+          }),
         },
       };
     }
     case 'manyWay': {
       return {
         $match: {
-          $expr: {
-            $in: ['$_id', '$$localAlias'],
-          },
+          $and: defaultMatches.concat({
+            $expr: {
+              $in: ['$_id', '$$localAlias'],
+            },
+          }),
         },
       };
     }
     case 'manyToMany': {
       if (assoc.dominant === true) {
         return {
           $match: {
-            $expr: {
-              $in: ['$_id', '$$localAlias'],
-            },
+            $and: defaultMatches.concat({
+              $expr: {
+                $in: ['$_id', '$$localAlias'],
+              },
+            }),
           },
         };
       }
 
       return {
         $match: {
-          $expr: {
-            $in: ['$$localId', `$${assoc.via}`],
-          },
+          $and: defaultMatches.concat({
```

Comment:
shouldn't we pass the $and inside the $expr to be consistent ?

Reply:
It wouldn't work as `populateQueries` are not meant to be used inside an $expr. (An object representing an expression must have only one field whereas `populateQueries.publicationState.live`, for example, has two)